### PR TITLE
Implementing accessibility

### PR DESCRIPTION
**Pull request by katia-utochkina**
## Description

This PR focuses on the app accessibility. It ensures that the app is written in semantic HTML. It adds missing attributes to SVG elements and ensures they are given accessible names in the HTML. It also makes sure that accessible names are provided to each ARIA input field . Apart from that, the PR ensures that users can navigate to all interactive interface components of the app  using a keyboard by giving focus styles to those elements that need them. It also makes that all visual information has a  contrast of at least 3:1 against the adjacent background.

## Acceptance Criteria
- [x] Semantically correct HTML
- [x] SVG elements have required attributes and accessible names
- [x] ARIA input fields have accessible names
- [x] Interactive interface components have focus styles
- [x] Visual information has a 3:1 contrast ration against the adjacent background

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Testing Steps / QA Criteria

* From your terminal, pull down this branch with `git pull origin refactoring` and check that branch out with `git checkout refactoring` 
* Then `npm start` to launch the app
* Check that the app is written using semantic HTML
* Check SVG elements have required attributes and accessible names 
* Check ARIA input fields have accessible names
* Check interactive interface components have focus styles
* Check visual information has a 3:1 contrast ration against the adjacent background

### CODE REVIEW

File: src/components/ListiView/Checkbox.js [hunk 1]
```diff
@@ -54,21 +54,33 @@ const Checkbox = ({ item, ariaLabel, index, shoppingList }) => {
...
         type="checkbox"
-        name={name}
-        id={name}
         className="list__item__input"
+        id={name}
+        name={name}
         value={name}
-        onChange={tickCheckBox}
         checked={checked}
-        aria-label={ariaLabel}
```
Comment (ciradu2204) [inline src/components/ListiView/Checkbox.js]: Why did you remove the arial-label here?

File: src/components/ListiView/Dialog.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const Dialog = ({ onCancel, onDelete }) => (
+  <>
+    <div className="dialog-wrapper" aria-hidden="true"></div>
+    <div id="dialog" role="dialog" aria-labelledby="dialog">
+      <div className="dialog__inner">
+        <p id="dialog-content">Are you sure you want to delete this item?</p>
+        <div className="btn-container">
+          <button onClick={onDelete} className="btn yes-button">
```
Comment (ciradu2204) [inline src/components/ListiView/Dialog.js]: Don't you think the cancel and yes button need ariallabels also?

File: src/components/ListiView/ListView.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React, { useState, useEffect } from 'react';
 
 import EmptyList from './EmptyList';
-import SearchForm from '../components/SearchForm';
```
Comment (ciradu2204) [inline src/components/ListiView/ListView.js]: I think you mispelled the name of the component. Doesn't it have to be `ListView` instead of `ListiView`. Also, I think the folders name should start with a lowercase.

**Review by ciradu2204 — APPROVED** [review]
Great job @katia-utochkina. I have left some few comments.

File: src/components/ListiView/Checkbox.js (hunk 1 above)
Comment (katia-utochkina) [reply src/components/ListiView/Checkbox.js]: Hi @ciradu2204, because for checkboxes, it's different, it's not needed there

File: src/components/ListiView/Dialog.js (hunk 2 above)
Comment (katia-utochkina) [reply src/components/ListiView/Dialog.js]: Buttons need aria-labels only if they use graphic instead of text

File: src/components/ListiView/ListView.js (hunk 3 above)
Comment (katia-utochkina) [reply src/components/ListiView/ListView.js]: Ha, good catch

File: public/index.html [hunk 4]
```diff
@@ -24,7 +24,7 @@
       work correctly both with client-side routing and a non-root public URL.
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
-    <title>React App</title>
+    <title>Smart Shopping App</title>
   </head>
```
Comment (faysvas) [inline public/index.html:28]: :+1:

**Review by faysvas — APPROVED** [review]
Great job on taking care of these last details! :100:
